app/knights/battle.py

```diff
@@ -0,0 +1,37 @@
+from app.knights.knights import lancelot, mordred, arthur, red_knight
+from app.config.config import KNIGHTS
+
+
+# BATTLE:
+class Battle:
+    @staticmethod
+    def battle(knights_config: dict) -> dict:
+        # 1 Lancelot vs Mordred:
+        lancelot.hp -= mordred.power - lancelot.protection
```

Comment:
The expression `lancelot.hp -= mordred.power - lancelot.protection` assumes that `lancelot` has a `protection` attribute. However, the `Knight` class does not define a `protection` attribute directly. You should calculate the total protection from the armour list as done in the `Knight` class initialization.